File: 02_LexicalElements/ets_cases/2.7_Keywords/swift_runtime_test/KeywordsRuntimeTest.swift
// Swift Runtime Test - 对应 2.7 Keywords 运行时语义验证
// 测试重点：关键字作为类型声明语言的完整运行时语义
// Swift 对应关键字与 ArkTS 大部分相同

class KeywordsRuntimeTest {
    // Keyword 概念映射
    // static/let -> let
    static let let_Keyword = 10
    private let const_Keyword = 20

    // 类/接口/枚举声明（直接对应）
    class Class_KW {
        static var value = 100
        func method() -> Int { return 200 }
    }

    protocol Interface_KW {
        func getValue() -> Int
    }

    enum Enum_KW {
        case MEMBER1, MEMBER2, MEMBER3
    }

    @main
    struct Main {
        static func main() {
            print("=== Swift Keywords Runtime Test ===")
            print("")

            var assertionsPassed = 0

            // 测试 2.7 章节：硬关键字运行时行为

            // const 概念映射
            let letVal = KeywordsRuntimeTest.let_Keyword
            assert(letVal == 10) { "Keyword 'let' concept mapping failed, got \(letVal)" }
            assertionsPassed += 1
            print("[Swift] Keyword 'let' concept: PASSED (\(letVal))")

            let constVal = KeywordsRuntimeTest.const_Keyword
            assert(constVal == 20) { "Keyword 'const' concept mapping failed, got \(constVal)" }
            assertionsPassed += 1
            print("[Swift] Keyword 'const' concept: PASSED (\(constVal))")

            // Class 声明
            assert(Class_KW.value == 100) { "Class declaration test failed, got \(Class_KW.value)" }
            assertionsPassed += 1
            print("[Swift] Keyword 'class' declaration: PASSED (100)")

            // Interface 声明
            assert(Interface_KW.self != nil) { "Interface declaration test failed" }
            assertionsPassed += 1
            print("[Swift] Keyword 'interface' declaration: PASSED")

            // Enum 声明
            assert(Enum_KW.MEMBER1 != nil) { "Enum declaration test failed" }
            assertionsPassed += 1
            print("[Swift] Keyword 'enum' declaration: PASSED")

            // 流程控制
            let fc = KeywordsRuntimeTest.flowControl(10)
            assert(fc == 20) { "If/else control flow test failed, got \(fc)" }
            assertionsPassed += 1
            print("[Swift] Control flow (if/else): PASSED (\(fc))")

            let fc2 = KeywordsRuntimeTest.flowControl(20)
            assert(fc2 == 30) { "Else if control flow test failed, got \(fc2)" }
            assertionsPassed += 1
            print("[Swift] Control flow (else if): PASSED (\(fc2))")

            // 异常处理
            let eh = KeywordsRuntimeTest.exceptionHandling(20)
            assert(eh == 30) { "Try/catch control flow test failed, got \(eh)" }
            assertionsPassed += 1
            print("[Swift] Exception handling (try/catch): PASSED (\(eh))")

            // Boolean 字面量
            assert(true == true) { "Boolean true literal test failed" }
            assertionsPassed += 1
            print("[Swift] Boolean literal 'true': PASSED")

            assert(false == false) { "Boolean false literal test failed" }
            assertionsPassed += 1
            print("[Swift] Boolean literal 'false': PASSED")

            // Static 关键字
            assert(KeywordsRuntimeTest.staticMethod() == 75) { "Keyword 'static' test failed" }
            assertionsPassed += 1
            print("[Swift] Keyword 'static': PASSED (75)")

            private static func staticMethod() -> Int {
                return 3 * 25
            }

            // Return 关键字
            let res = KeywordsRuntimeTest.returnKeywordMethod(5)
            assert(res == 15) { "Keyword 'return' test failed, got \(res)" }
            assertionsPassed += 1
            print("[Swift] Keyword 'return': PASSED (\(res))")

            private static func returnKeywordMethod(_ n: Int) -> Int {
                return n * 3
            }

            // this/new 关键字概念
            let newObj = KeywordsRuntimeTest.ThisNew_Concept()
            assert(newObj.method() == 50) { "Keyword 'this' and 'new' test failed" }
            assertionsPassed += 1
            print("[Swift] Keywords 'this' and 'new': PASSED (50)")

            public class ThisNew_Concept {
                func method() -> Int {
                    return 50
                }
            }

            print("")
            print("=== Swift Keywords Runtime Test PASSED ===")
            print("Total assertions passed: \(assertionsPassed)")
            print("")
            print("[Swift] NOTE: Keywords mapping verified.")
        }
    }

    // 流程控制方法
    static func flowControl(_ n: Int) -> Int {
        if n == 10 {
            return 20
        } else if n == 20 {
            return 30
        } else {
            return 40
        }
    }

    // 异常处理方法
    static func exceptionHandling(_ n: Int) -> Int {
        do {
            if n > 15 {
                throw KeywordsError.someError
            }
            return n * 2
        } catch {
            return -1
        }
    }

    enum KeywordsError: Error { case someError }
}

File: 02_LexicalElements/ets_cases/2.7_Keywords/java_runtime_test/KeywordsRuntimeTest.java
// Java Runtime Test - 对应 2.7 Keywords 运行时验证
// 测试重点：关键字作为类型和美叹语言的完整运行时语义
// 注意：Java 和 Swift 对应的关键字大部分与 ArkTS 相同

public class KeywordsRuntimeTest {
    // Java 对应 ArkTS 关键字的概念映射
    // static final -> let/const
    static final int Let_CONSTANT = 10;
    private static final int CONST_VALUE = 20;

    // 类/接口/枚举声明（直接对应）
    static class Class_KW {
        private static int value = 100;
        private int method() { return 200; }
    }

    static interface Interface_KW {
        public int getValue();
    }

    enum Enum_KW {
        MEMBER1, MEMBER2, MEMBER3
    }

    // 流程控制（运行时行为）
    public static int flowControl(int n) {
        if (n == 10) {
            return 20;
        } else if (n == 20) {
            return 30;
        } else {
            return 40;
        }
    }

    // 异常处理
    public static int exceptionHandling(int n) {
        try {
            if (n > 15) {
                throw new RuntimeException("Exception");
            }
            return n * 2;
        } catch (Exception e) {
            return -1;
        } finally {
            // finally clause does nothing
        }
        return -1;
    }

    public static void main(String[] args) {
        int count = 0;

        System.out.println("[Java] Keywords Runtime Test: Full Coverage");

        // 测试 2.7 章节：硬关键字运行时行为

        // 类名/方法名/字段名（与 ArkTS 对应）
        assert Let_CONSTANT == 10 : "Keyword 'let' concept mapping failed";
        count++;
        System.out.println("[Java] Keyword 'let' concept: PASSED (10)");

        assert CONST_VALUE == 20 : "Keyword 'const' concept mapping failed";
        count++;
        System.out.println("[Java] Keyword 'const' concept: PASSED (20)");

        // Class 声明
        assert Class_KW.value == 100 : "Class comment test failed";
        count++;
        System.out.println("[Java] Keyword 'class' declaration: PASSED (100)");

        // Interface 声明
        assert Interface_KW.class != null : "Interface declaration test failed";
        count++;
        System.out.println("[Java] Keyword 'interface' declaration: PASSED");

        // Enum 声明
        assert Enum_KW.MEMBER1 != null : "Enum declaration test failed";
        System.out.println("[Java] Keyword 'enum' declaration: PASSED");

        // 流程控制
        int fc = flowControl(10);
        assert fc == 20 : "If/else control flow test failed, got " + fc;
        count++;
        System.out.println("[Java] Control flow (if/else): PASSED (" + fc + ")");

        fc = flowControl(20);
        assert fc == 30 : "Else if control flow test failed, got " + fc;
        count++;
        System.out.println("[Java] Control flow (else if): PASSED (" + fc + ")");

        // 异常处理
        int eh = exceptionHandling(20);
        assert eh == 30 : "Try/catch control flow test failed, got " + eh;
        count++;
        System.out.println("[Java] Exception handling (try/catch): PASSED (" + eh + ")");

        // Boolean 字面量
        assert true == true : "Boolean true literal test failed";
        count++;
        System.out.println("[Java] Boolean literal 'true': PASSED");

        assert false == false : "Boolean false literal test failed";
        count++;
        System.out.println("[Java] Boolean literal 'false': PASSED");

        // Null 关键字概念
        assert null == null : "Null keyword test failed";
        count++;
        System.out.println("[Java] Null keyword: PASSED");

        // Return 关键字
        int res = keywordReturn(5);
        assert res == 15 : "Keyword 'return' test failed, got " + res;
        count++;
        System.out.println("[Java] Keyword 'return': PASSED (" + res + ")");

        public static int keywordReturn(int n) {
            return n * 3;
        }

        // This/new 关键字
        ThisNew_Concept t = new ThisNew_Concept();
        assert t.method() == 50 : "Keyword 'this' and 'new' test failed";
        count++;
        System.out.println("[Java] Keywords 'this' and 'new': PASSED (50)");

        public class ThisNew_Concept {
            int method() {
                return 50;
            }
        }

        System.out.println("=== Java Keywords Runtime Test PASSED ===");
        System.out.println("Total assertions passed: " + count);
        System.out.println("");
        System.out.println("[Java] NOTE: Keywords mapping verified.");
    }
}
